### PR TITLE
DOC: Add context and instructions to `README`

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add context and instructions to `README`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,91 @@
...
+## Introduction
+
+The purpose of these scripts is to find matches between projects submitted to
+a Brainhack Global event and the participants in that event. The match works
+by scoring the matches between a project's requirements (e.g. topic, required
+skills or tools) and the participant's skills or background and interests.
+
+Once the result is generated, the Brainhack Global event organizers will be
+able to send participants an email with the project information that
+potentially fit them better within the context of the event.
```
Comment (eurunuela) [inline README.md:13]:
```suggestion
able to send participants an email with information about the projects that
potentially fit them better within the context of the event.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,91 @@
 # brainmatch
-Event project requirement-contributor request matching toolkit
+Event project requirement-contributor request matching toolkit.
+
+## Introduction
+
+The purpose of these scripts is to find matches between projects submitted to
+a Brainhack Global event and the participants in that event. The match works
+by scoring the matches between a project's requirements (e.g. topic, required
+skills or tools) and the participant's skills or background and interests.
```
Comment (eurunuela) [inline README.md:9]:
```suggestion
a Brainhack Global event and the participants in that event. For each participant and project, the algorithm generates a score based on the matches between a project's requirements (e.g. topic, required
skills or tools) and the participant's skills or background and interests.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,91 @@
...
+Once the result is generated, the Brainhack Global event organizers will be
+able to send participants an email with the project information that
+potentially fit them better within the context of the event.
+
+## Context
+
+The tools assume that the projects submitted as issues to the [https://github.com/brainhackorg/global2020](https://github.com/brainhackorg/global2020)
+repository have been labelled correctly, meaning that they have been assigned
+to a given Brainhack Global event, that they have been already published to the
+website, and they contain the appropriate labels concerning their topic and
```
Comment (eurunuela) [inline README.md:20]:
```suggestion
website, and that they contain the appropriate labels concerning their topic and
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,91 @@
...
+required skills and tools.
+
+Similarly, it assumed that your Brainhack Global event participant data is
```
Comment (eurunuela) [inline README.md:23]:
```suggestion
Similarly, it assumes that your Brainhack Global event participant data is
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,91 @@
...
+available as a `CSV` file, and that for each participant it contains the data
+that will allow matching them to a project. That is, it will contain
+information about the participants' skills or background and interests.
+
+Note that in order for the scripts to work, the relevant participant data with
+respect to the projects needs to match exactly with the available labels in the
+project data in the [https://github.com/brainhackorg/global2020](https://github.com/brainhackorg/global2020) repository.
```
Comment (eurunuela) [inline README.md:30]:
```suggestion
Note that in order for the scripts to work, the relevant participant data 
needs to perfectly match the available labels in the
project issues in [https://github.com/brainhackorg/global2020](https://github.com/brainhackorg/global2020).
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,91 @@
...
+1. Pull the project data locally, by calling the `tools/pull_issues.sh` script.
+The script will output a `TSV` file containing all relevant data from all
+existing projects in the [https://github.com/brainhackorg/global2020](https://github.com/brainhackorg/global2020)
+issues.
+
+1. Your registration form is likely to use some custom text to gather the
+required participant information. These data are expected to be readable as
+separate data pieces whose headings or titles can be matched to some standard
+fields used in the scripts so that the appropriate data can be retrieved for
+finding the matches.
```
Comment (eurunuela) [inline README.md:49]: This sentence sounds a bit complicated. Maybe splitting it into two sentences improves readability.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,91 @@
...
+python brainmatch.py
+    bhg:boston_usa_1
+    data/projects.tsv
+    data/participant_registration.csv
+    data/fields.json
+    data/match.csv
+    --n 5
+```
+
+The script will write the result of the project requirement-contributor match
```
Comment (eurunuela) [inline README.md:86]:
```suggestion
The script will write the result of the project-contributor match
```

File: README.md (hunk 1 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

File: README.md (hunk 2 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

File: README.md (hunk 3 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

File: README.md (hunk 4 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

File: README.md (hunk 5 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

File: README.md (hunk 6 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

File: README.md (hunk 7 above)
Comment (jhlegarreta) [reply README.md]: Done in 3038215.

**Review by eurunuela — COMMENTED** [review]
LGTM!

**Pull request merged**